pkgs/development/compilers/openjdk/bootstrap.nix

```diff
@@ -40,13 +41,13 @@ let
     LIBDIRS="$(find $out -name \*.so\* -exec dirname {} \; | sort | uniq | tr '\n' ':')"
 
     for i in $out/bin/*; do
-      patchelf --set-interpreter ${glibc.out}/lib/ld-linux*.so.2 $i || true
-      patchelf --set-rpath "${glibc.out}/lib:$LIBDIRS" $i || true
+      patchelf --set-interpreter ${stdenv.cc.libc}/lib/ld-linux*.so.2 $i || true
+      patchelf --set-rpath "${stdenv.cc.libc}/lib:$LIBDIRS" $i || true
     done
 
     find $out -name \*.so\* | while read lib; do
-      patchelf --set-interpreter ${glibc.out}/lib/ld-linux*.so.2 $lib || true
-      patchelf --set-rpath "${glibc.out}/lib:${stdenv.cc.cc.lib}/lib:$LIBDIRS" $lib || true
+      patchelf --set-interpreter ${stdenv.cc.libc}/lib/ld-linux*.so.2 $lib || true
```

Comment:
Not sure whether `glibc` can be dropped like that.

Reply:
There are nss modules in glibc, but musl don't support those anyway.
On at least NixOS we also use nscd, which is apparently also supported by musl. 
I have not seen any port of jdk, that uses anything but glibc though.

Reply:
While we are touching this wouldn't it make sense to use `$NIX_CC/nix-support/dynamic-linker` instead of `ld-linux*.so.2`?

In my yet not wroking WIP for aarch64 for openjdk I had to do that. Due to time contraints that work might not continue soon on my end thought..

Reply:
Thanks for the info you two!

Yeah my thoughts are even if it doesn't work, its better to require `stdenv.cc.libc` to be `glibc` (which could be achieved with a one-off override, rather than whole-world stdenv switch) than hack it in like that.